src/server.ts

```diff
@@ -0,0 +1,232 @@
+
+import { Children, createElement } from 'react';
+import * as ReactDOM from 'react-dom/server';
+import ApolloClient from 'apollo-client';
+import flatten = require('lodash.flatten');
+import assign = require('object-assign');
+
+/*
+
+React components can return a `falsy` (null, false) value,
+representation of a native DOM component (such as <div /> or React.DOM.div())
+or another composite component. Components can have a render function (for components).
+They can also pass through children which we want to analyze as well.
+
+To get data from `connect()` components we do a few things:
+
+1. if ssr is not falsy, move the query to a place to batch call it
+
+Ideally, we go through the tree and find all `connect()`s (recursively going through tree)
+If we reach the end of all nodes, we kick off the queries. Once queries have returned,
+we try to go through their children components again to see if we discover any
+more queries. Then once we reach th end, we render the dom.
+
+We recursively do this until the tree is done.
+
+So! Given a component:
+
+1. See if it is falsy (end of line)
+2. Bulid the context and props (global props + defaultProps + parent props)
+3. See if the component is a `connect()`
+3a. Get the queries using props + state
+3b. as long as ssr != false, pass the query to the array to be called
+4. Create the component (or child if connect) (`componentWillMount` will run)
+5. Render the component
+6. Repeat
+
+*/
+
+declare interface Context {
+  client?: ApolloClient;
+  store?: any;
+  [key: string]: any;
+}
+
+declare interface QueryTreeArgument {
+  components: any;
+  queries?: any[];
+  defaultProps?: any;
+  context?: Context;
+}
+
+const defaultReactProps = {
+  loading: true,
+  errors: null,
+};
+
+
+export function getPropsFromChild(child, defaultProps = {}) {
+  const { props, type } = child;
+  let ownProps = assign(defaultProps, props);
+  if (type && type.defaultProps) ownProps = assign(defaultProps, type.defaultProps, props);
```

Comment:
pretty much yes, if I'm reading this correctly: https://github.com/facebook/react/blob/49238b944046abf086ba80ab5e984ce8a8175a1d/src/isomorphic/classic/element/ReactElement.js#L207-L214